src/parse/converters/expressions/primary/literal/objectLiteral/keyValuePair.js

```diff
@@ -10,6 +10,8 @@ export default function readKeyValuePair ( parser ) {
 	// allow whitespace between '{' and key
 	parser.allowWhitespace();
 
+	const refKey = parser.nextChar() !== '\'' && parser.nextChar() !== '"';
```

Comment:
Why wouldn't it? `4` is a valid key. Also see https://github.com/ractivejs/ractive/blob/59686405a594eb239c7aec7954bfccaebd200f34/src/parse/converters/expressions/shared/readKey.js#L16